### PR TITLE
Update Batch DataPlane SDK to support new REST API version

### PR DESCRIPTION
**Pull request by matthchr**
- Added support for new 2020-03-01 REST API version for batch dataplane

This release is for JEDI.

### CODE REVIEW

**Comment by matthchr** [comment]
I need to run a few more live tests before merge, but want to open up to feedback from folks because this release is for JEDI so timelines are tight.

Please do not merge this until I remove the `do not merge` tag.

File: sdk/batch/Microsoft.Azure.Batch/CHANGELOG.md
Comment (AlexGhiondea) [inline sdk/batch/Microsoft.Azure.Batch/CHANGELOG.md]: @chidozieononiwu should we have the (Unreleased) moniker here?

Comment (chidozieononiwu) [reply sdk/batch/Microsoft.Azure.Batch/CHANGELOG.md]: Yes, @Alex  You are right. One can just run the Update-Changelog script
> To set version it with Unreleased moniker. In most cases the build pipeline should handle this.
`& "eng/common/Update-Change-Log.ps1" -Version '13.0.0' -ChangeLogPath 'sdk/batch/Microsoft.Azure.Batch/CHANGELOG.md'`

> To set version when preparing for release, with the current date in the header
`& "eng/common/Update-Change-Log.ps1" -Version '13.0.0' -ChangeLogPath 'sdk/batch/Microsoft.Azure.Batch/CHANGELOG.md' -Unreleased false -ReplaceVersion true` to set the  current date in the changelog header

File: sdk/batch/Microsoft.Azure.Batch/src/Microsoft.Azure.Batch.csproj
Comment (AlexGhiondea) [inline sdk/batch/Microsoft.Azure.Batch/src/Microsoft.Azure.Batch.csproj]: Do you want to release a preview first?

**Review by AlexGhiondea — APPROVED** [review]

File: sdk/batch/Microsoft.Azure.Batch/CHANGELOG.md
Comment (matthchr) [reply sdk/batch/Microsoft.Azure.Batch/CHANGELOG.md]: Does the tooling automatically fix that once it goes live? If so, cool!

File: sdk/batch/Microsoft.Azure.Batch/src/Microsoft.Azure.Batch.csproj
Comment (matthchr) [reply sdk/batch/Microsoft.Azure.Batch/src/Microsoft.Azure.Batch.csproj]: No preview is needed.

File: sdk/batch/Microsoft.Azure.Batch/CHANGELOG.md
Comment (chidozieononiwu) [reply sdk/batch/Microsoft.Azure.Batch/CHANGELOG.md]: After a release there will be a PR to update to the next version. But you should run it before releasing so that it sets the date. Or you can also add the date manually. just ,make sure you stick to the [changelog guidance](https://github.com/Azure/azure-sdk/blob/master/docs/policies/releases.md#changelog-guidance)

Comment (matthchr) [reply sdk/batch/Microsoft.Azure.Batch/CHANGELOG.md]: I fixed this.

File: eng/mgmt/mgmtmetadata/batch_data-plane.txt
Comment (xingwu1) [inline eng/mgmt/mgmtmetadata/batch_data-plane.txt]: What is this?

**Review by xingwu1 — APPROVED** [review]

File: eng/mgmt/mgmtmetadata/batch_data-plane.txt
Comment (matthchr) [reply eng/mgmt/mgmtmetadata/batch_data-plane.txt]: It's a file automatically created by autorest

**Pull request merged**
